### ISSUE TITLE
Advanced Portals Forget Name on Re-Login

### ISSUE THREAD

**Issue by MacharaStormwing** [RandyKnapp/ValheimMods#617]
Since the latest update supporting Hildir's Request update whenever I start the game again or logout and login the name of
Advanced Portals is set to Empty.
This causes those Portals to not show up in the Portal Selection List. I have to give it a name again for it to show up in the portal selection lists. The mod keeps working during the session but again upon restarting or re-logging the name goes missing again.
I use this mod in cobination with X Portals.

Such portals that have missing name still show as connected but after giving it a name it no longer is connected.

If you need further information feel free to ask.

**Comment by SpikeHimself** [comment]
To the Advanced Portals devs: the cause of this issue is that Advanced Portals are not returned by `ZDOMan.instance.GetPortals()`

**Comment by kmGaijin** [comment]
Reporting this issue here as well.

**Comment by Vapok** [comment]
I swear this was working before.  I'll get this fixed up.

**Comment by SpikeHimself** [comment]
Thank you @Vapok!

I think a recent update (a minor one after Hildir's Request), changed how ZDOIDs are recycled. Or rather, it added the fact that they can be recycled. I had a similiar issue in XPortal where old portal targets no longer existed when starting a new session, because the ZDOIDs did not persist between sessions.

To solve that, I save the "current" ZDOID of each portal in a ZDO property I called "PreviousId", so that I can search for it later. But that's another story and probably not relevant to the issue at hand :)

**Comment by Vapok** [comment]
@SpikeHimself Apologies on this taking so long. I went down a rabbit hole of trying to use the patch that was provided by Blaxxun, but it seems to now be out of date.  Either way, looks like 0.217.14 added the ability to add portal prefab hashes during the Game.Awake. So now I'm adding the prefab hashes at Game.Awake and completely removes the need to patch ZDOMan at all.

This is now released as version 1.0.8 of Advanced Portals
